crates/bevy_ecs/src/schedule_v3/schedule.rs

```diff
@@ -0,0 +1,1022 @@
+use std::{
+    borrow::Cow,
+    fmt::{Debug, Write},
+    result::Result,
+};
+
+use bevy_utils::{
+    petgraph::{algo::tarjan_scc, prelude::*},
+    thiserror::Error,
+    tracing::{error, info, warn},
+    HashMap, HashSet,
+};
+
+use fixedbitset::FixedBitSet;
+
+use crate::{
+    self as bevy_ecs,
+    component::ComponentId,
+    schedule_v3::*,
+    system::{BoxedSystem, Resource},
+    world::World,
+};
+
+/// Resource that stores [`Schedule`]s mapped to [`ScheduleLabel`]s.
+#[derive(Default, Resource)]
+pub struct Schedules {
+    inner: HashMap<BoxedScheduleLabel, Schedule>,
+}
+
+impl Schedules {
+    /// Constructs an empty `Schedules` with zero initial capacity.
```

Comment:
What about `get_many_mut`? Exposing that would probably be more convenient.

Reply:
I may have been thinking about these API's wrong. Let me start over and ask a more general question. What I would like to do is something like this:

```rust
fn my_exclusive_system() {
  let resource = world.get_resource::<MyResource>();

  if MyResource == some_value {
    // this won't work because world is already borrowed by resource.
    world.run_schedule(MySchedule);
  } else {
    world.run_schedule(MyOtherSchedule);
  }
}
```

Writing this code down made me realize that the problem isn't really solvable with changing the API's. If you want to run a schedule on the world there can't be any active borrows into it. So this would be cloning/copying data out, using a resource scope or removing the data from the world, which we can do with existing API's.